api/models/ssl.go

```diff
@@ -273,14 +234,136 @@ func ListSSLs(a string) (SSLs, error) {
 	return ssls, nil
 }
 
+func UpdateSSL(app, process string, port int, body, key string, chain string) (*SSL, error) {
+	a, err := GetApp(app)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// validate app is not currently updating
+	if a.Status != "running" {
+		return nil, fmt.Errorf("can not update app with status: %s", a.Status)
+	}
+
+	// store old cert name
+	oldCertName := certName(app, process, port)
+
+	// validate process exists
+	if oldCertName == "" {
+		return nil, fmt.Errorf("no certificate configured for %s port %d", process, port)
+	}
+
+	outputs := a.Outputs
+
+	balancer := outputs[fmt.Sprintf("%sPort%dBalancerName", UpperName(process), port)]
+
+	if balancer == "" {
+		return nil, fmt.Errorf("Balancer ouptut not found. Please redeploy your app and try again.")
+	}
+
+	// upload new cert
+	arn, err := uploadCert(a, process, port, body, key, chain)
+
+	if err != nil {
+		return nil, err
+	}
+
+	input := &elb.SetLoadBalancerListenerSSLCertificateInput{
+		LoadBalancerName: aws.String(balancer),
+		LoadBalancerPort: aws.Int64(int64(port)),
+		SSLCertificateId: aws.String(arn),
+	}
+
+	// change cert on listener
+	operation := func() error {
+		_, err = ELB().SetLoadBalancerListenerSSLCertificate(input)
+
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	err = backoff.Retry(operation, backoff.NewExponentialBackOff())
```

Comment:
Yes. If I'm understanding correctly, the default exponential algorithm I've used here has a max time of 15 minutes: https://github.com/cenkalti/backoff/blob/master/exponential.go#L8-L94

It's possible to tweak the settings to achieve different curves and maxes.